[doc] Improve description of `valid_systems` and `valid_prog_environs` in the tutorial and other minor tutorial fixes (#3208)
reframe-hpc/reframe: docs/tutorial.rst

    @rfm.simple_test
    class stream_test(rfm.RunOnlyRegressionTest):

>         # ...

Author: Oh okay, I thought it was invalid.

   self.valid_prog_environs = ['+stream']

> This tells ReFrame that this test is valid only for environments that define the ``stream`` feature; the ``+`` prefix requests a feature. The syntax defined in :attr:`valid_prog_environs` allows for other requirements.

Reviewer: ```suggestion
This tells ReFrame that this test is valid only for environments that define the ``stream`` feature (the ``+`` prefix requests a feature; we will explain the syntax later in this section).
```

And then expand later in the bullet list that talks about features and extras (I can do that).
